bindings/java/java/src/main/java/org/ethereum/evmc/EvmcVm.java

```diff
@@ -14,30 +14,20 @@
  * <p>Defines the Java methods capable of accessing the evm implementation.
  */
 public final class EvmcVm implements AutoCloseable {
-  private static EvmcVm evmcVm;
-  private static boolean isEvmcLibraryLoaded = false;
   private ByteBuffer nativeVm;
+
+  static {
+    System.loadLibrary("evmc-java");
```

Comment:
Instead of this, we need copy the shared library out of the JAR into some temporary directory and load from there. See https://github.com/ethereum/evmc/pull/532#discussion_r497421023